sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java

```diff
@@ -41,6 +41,7 @@
     public CommunicationTokenCredential(String token) {
         Objects.requireNonNull(token, "'token' cannot be null.");
         setToken(token);
+        refreshTimeBeforeTokenExpiry = CommunicationTokenRefreshOptions.getDefaultRefreshTimeBeforeTokenExpiry();
```

Comment:
Wouldn't it be possible to compose the `CommunicationTokenRefreshOptions` object here and drop the call to `getDefaultRefreshTimeBeforeTokenExpiry()`? We could then make the `public static Duration getDefaultRefreshTimeBeforeTokenExpiry()` private and simplify the public API.